File: src/ts/day1.ts

//@ts-ignore
import * as path from "path";
//@ts-ignore
import { readFileSync } from "fs";
        
//@ts-ignore
const input: string = readFileSync(path.join(__dirname, "../../data/day1.prod"), "utf8");

let list: number[] = input.split("\n\n")
            .map(x => x.split("\n")
                 .map(x => {  
                     let numb: number = parseInt(x);
                     return !isNaN(numb)? numb : 0;
                 }).reduce((sum: number, curVal: number) => sum + curVal, 0)
                ).sort((x,y) => x-y);

console.log("=== Part 1 ===");

console.log(list.slice(-1));

console.log("=== Part 2 ===");
console.log(list.slice(-3).reduce((sum, val)=>sum+val, 0));

File: src/bin/day1.rs
fn main() {
    let mut sorted_calories = include_str!("../../data/day1.prod").split("\n\n")
        .map(|x| {
            return x.split('\n').flat_map(str::parse::<usize>)
                .sum::<usize>();
        }).collect::<Vec<usize>>();

    println!("Part 1 => {:?}", sorted_calories.iter().max().unwrap());

    sorted_calories.sort_by(|a, b| b.cmp(a));

    println!("Part 2 => {:?}\n", sorted_calories.iter().take(3).sum::<usize>());
}

